vnext/Desktop/Modules/BlobModule.cpp

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "BlobModule.h"
+
+#include <unicode.h>
+
+// React Native
+#include <cxxreact/Instance.h>
+#include <cxxreact/JsArgumentHelpers.h>
+#include <cxxreact/ModuleRegistry.h>
+
+// Windows API
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Security.Cryptography.h>
+
+using namespace facebook::xplat;
+
+using folly::dynamic;
+using std::lock_guard;
+using std::mutex;
+using std::string;
+using std::vector;
+using std::weak_ptr;
+using winrt::Windows::Foundation::GuidHelper;
+using winrt::Windows::Security::Cryptography::CryptographicBuffer;
+
+namespace {
+constexpr char moduleName[] = "BlobModule";
+constexpr char blobURIScheme[] = "blob";
+
+weak_ptr<Microsoft::React::IWebSocketModuleContentHandler> s_contentHandler;
+} // namespace
+
+namespace Microsoft::React {
+
+BlobModule::BlobModule() noexcept {
+  m_contentHandler = std::make_shared<BlobWebSocketModuleContentHandler>();
+  s_contentHandler = m_contentHandler;
```

Comment:
>s_contentHandler [](start = 2, length = 16)

Should we have checks that only a single one is initialized? So that GetInstance() is guaranteed to always return the same value?

Reply:
Good catch.

Circumstantially, this shouldn't be a problem because these modules are singletons. This constructor should only be called once.

Reply:
If we have multiple react instances, then the constructor will be called per each instance. Could we avoid using global variables? A better design would be to use the NativeModules 2.0 and keep any required variables per React Insatnce in the ReactContext::Properties().

---
In reply to: [494032163](https://github.com/microsoft/react-native-windows/pull/6079#discussion_r494032163) [](ancestors = 494032163)